TrkFilters/src/HelixFilter_module.cc

```diff
@@ -292,9 +296,35 @@ namespace mu2e
         }
       }
     }
+
+    bool dt_range = false;
+    if (_maxDt0 < 0){
+      dt_range = true;
+    }
+    else{
+      for(auto ihs = hscol->begin();ihs != hscol->end(); ++ihs) {
+        auto const&  hel0 = ihs;
+        float  hel0_t0 =  hel0->t0()._t0;
+
+        for(auto ihs = hscol->begin();ihs != hscol->end(); ++ihs) {
```

Comment:
This loops over the entire collection again (reusing the `ihs` variable name, perhaps use `jhs` or something?), can this find the same helix again which would have `dt == 0` (i.e. `hel0 == hel1`). Perhaps this loop could start at the next iteration, avoiding this collision as well as rechecking comparisons you've already done, something like:
`for(auto jhs = std::next(ihs); jhs != hscol->end(); ++jhs) {`